### DISCUSSION TITLE
Support for ETag and pagination link

### DISCUSSION THREAD

**Discussion by Biplab-Dutta** [forem/forem#17807] [Feature Requests]
I've been trying to build something out of the DEV API but after reading the documentation and playing around with the API a bit, I soon came to realize that there are some headers missing in the API response.

- **Support for ETag**: Most of the server today returns an ETag in their response while performing a Get request. ETag helps a lot in implementing caching mechanisms and avoiding unwanted logic in the client-side production codebase.
- **Pagination Link**: I've seen that the Dev API supports pagination but there is no information about the maximum number of pages in a GET request. Having some information regarding the maximum number of pages in the response can help the client-side developers when implementing the pagination feature.

I am sharing a screenshot of GitHub's API response. They have Etag and Link headers for ETag value and information on the maximum number of pages respectively. Something similar could also be implemented in the DEV's API.

![Screenshot from 2022-06-02 15-08-14](https://user-images.githubusercontent.com/63902683/171605614-f7d92617-eef1-4561-ad0e-bc68336da51a.png)

I highly suggest the Dev API Developers take a look at GitHub's API doc regarding [ETag](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#conditional-requests) and [Link](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#link-header) heders for further information.

**Comment by Isaias2513** [comment]
As the screenshoot is clarifying your issue. Same like you yesterday I was checking the api connection of [Drywall installation Boston](https://smartchoiceremodel.com/drywall-services/) business so that was also showing the same issue. But I added the header through custom coding while searching from youtube. Which helped me in fixing the issue. As you will find the header script by searching.
